mlrun/runtimes/sparkjob/spark3job.py

```diff
@@ -334,14 +405,68 @@ def with_executor_node_selection(
                 "Setting node name is not supported for spark runtime"
             )
         if affinity:
-            raise NotImplementedError(
-                "Setting affinity is not supported for spark runtime"
-            )
+            self.spec.executor_affinity = affinity
         if node_selector:
             self.spec.executor_node_selector = node_selector
         if tolerations:
             self.spec.executor_tolerations = tolerations
 
+    def with_preemption_mode(
+        self, mode: typing.Union[mlrun.api.schemas.function.PreemptionModes, str]
+    ):
+        """
+        Preemption mode controls whether pods can be scheduled on preemptible nodes.
+        Tolerations, node selector, and affinity are populated on preemptible nodes corresponding to the function spec.
+
+        Three modes are supported:
+
+        * **allow** - The function can be scheduled on preemptible nodes
+        * **constrain** - The function can only run on preemptible nodes
+        * **prevent** - The function cannot be scheduled on preemptible nodes
+
+        :param mode: accepts allow | constrain | prevent defined in :py:class:`~mlrun.api.schemas.PreemptionModes`
```

Comment:
I think the docstring for this one should be like the exception - pointing on the right methods to use